### PR TITLE
Add Search Bar Widget

### LINKED ISSUE #33: [REQUEST] Adding a search bar
Hello!

I've been recently using glance as my dashboard what can I say other than it's AWESOME!

I really love the aesthetics and I would love for a search bar that leads to a search engine.

This is pretty amazing for a V1 projects!

Thanks in advance.

### PR DESCRIPTION
**Pull request by chand1012**
Adds a simple search bar widget that allows you to specify a search engine. Also adds the proper documentation for use. Closes #33.

search-widget-preview

### CODE REVIEW

**Comment by gitsang** [comment]
I compiled and test a version locally, is great.

But I would like to raise a few minor requests:

1. Can it automatically focus on the first search box? (Perhaps this should not be solved by the widget, I am not very familiar with frontend implementation.)
2. Is it possible to add a height adjustment option? Or support adding multiple search boxes in a single widget?

**Comment by gitsang** [comment]
Oh, and the query parameter doesn't seem to be working.

image

image

File: internal/widget/search.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+func (widget *Search) Initialize() error {
+	widget.withTitle("Search").withError(nil)
+
+	if widget.SearchURL == "" {
+		// set to the duckduckgo search engine
+		widget.SearchURL = "https://duckduckgo.com/?q="
+	}
+
+	// if no query is provided, leave an empty string
```
Comment (gitsang) [inline internal/widget/search.go:23]: Is here not being handled?

**Comment by svilenmarkov** [comment]
Hey, thanks for contributing!

I made a couple of changes and added a few more features, hope you like them!

**Pull request merged**